### PR TITLE
[Documentation] DTI::DTI library perldocification

### PR DESCRIPTION
**Pull request by cmadjar**
Using perldoc/perlpod to document `DTI.pm` so that users can type in the terminal
`perldoc DTI.pm` and get the documentation.

Using the `pod2markdown` tool from perl `Pod::Markdown` module, the `DTI.md` file has been created so that we have a web displayed version of the documentation.
`pod2markdown DTI.pm > DTI.md`

Dependencies added: perldoc, Pod::Markdown, Pod::Usage

### CODE REVIEW

File: DTIPrep/DTI/DTI.pm [hunk 1]
```diff
@@ -562,16 +672,25 @@ sub insertMincHeader {
...
-Outputs: - 1 if all processed info were inserted in the processed minc header
-         - undef otherwise
+
+=head3 insertProcessInfo($raw_dti, $data_dir, $processed_minc, ...)
+
+This will insert in the header of the processed file processing information. If
+one of the value to insert is not defined, return undef, otherwise return 1.
+
+INPUT:
+  - $raw_dti       : raw DTI MINC file to grep header information
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: ..header information _from_.

File: DTIPrep/DTI/DTI.pm [hunk 2]
```diff
@@ -609,18 +728,22 @@ sub insertProcessInfo {
 }
 
 
+=pod
+
+=head3 insertAcqInfo($raw_dti, $processed_minc)
 
+Insert acquisition information extracted from raw DTI dataset and insert it in
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: _Inserts_...

File: DTIPrep/DTI/DTI.pm [hunk 3]
```diff
@@ -644,18 +767,23 @@ sub insertAcqInfo {
 }
 
 
+=pod
 
+=head3 insertFieldList($raw_dti, $processed_minc, $minc_field)
 
+Insert information extracted from raw DTI dataset and insert it in the
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: _Inserts_...

File: DTIPrep/DTI/DTI.pm [hunk 4]
```diff
@@ -644,18 +767,23 @@ sub insertAcqInfo {
...
+processed file. If one of the value to insert is not defined, return undef,
+otherwise return 1.
+
+INPUT:
+  - $raw_dti       : raw DTI MINC file to grep header information
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: ..information _from_.

File: DTIPrep/DTI/DTI.pm [hunk 5]
```diff
@@ -689,16 +817,24 @@ sub insertFieldList {
...
-Inputs:  - $argument: argument to be inserted in minc header
-         - $value: value of the argument to be inserted in minc header
-         - $minc: minc file
-         - $awk: awk information to check if argument not already inserted in minc header
-Outputs: - 1 if argument was indeed inserted into the minc file
-         - undef otherwise
+
+=head3 modify_header($argument, $value, $minc, $awk)
+
+Function that runs C<minc_modify_header> and insert MINC header information if
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: and inserts

File: DTIPrep/DTI/DTI.pm [hunk 6]
```diff
@@ -689,16 +817,24 @@ sub insertFieldList {
...
+not already inserted.
+
+INPUT:
+  - $argument: argument to be inserted in MINC header
+  - $value   : value of the argument to be inserted in MINC header
+  - $minc    : MINC file
+  - $awk     : awk info to check if argument inserted in MINC header
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: if argument was inserted

File: DTIPrep/DTI/DTI.pm [hunk 7]
```diff
@@ -719,16 +855,22 @@ sub modify_header {
 }
 
 
+=pod
 
+=head3 fetch_header_info($field, $minc, $awk, $keep_semicolon)
+
+Function that fetch header information in MINC file.
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: that fetches

File: DTIPrep/DTI/DTI.pm [hunk 8]
```diff
@@ -746,18 +888,22 @@ sub fetch_header_info {
 }
 
 
+=pod
 
+=head3 get_header_list($splitter, $fields)
 
+Get the list of arguments and values to insert into the MINC header
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: Gets

File: DTIPrep/DTI/DTI.pm [hunk 9]
```diff
@@ -746,18 +888,22 @@ sub fetch_header_info {
...
+(C<acquisition:*>, C<patient:*> and C<study:*>).
 
+INPUT:
+  - $splitter: splitter used to split list of fields stored in C<$fields>
```
Comment (nicolasbrossard) [inline DTIPrep/DTI/DTI.pm]: I would use delimiter instead of splitter

File: DTIPrep/DTI/DTI.pm (hunk 3 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 4 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 5 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 6 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 7 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 8 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done

File: DTIPrep/DTI/DTI.pm (hunk 9 above)
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: cool!

File: DTIPrep/DTI/DTI.pm [hunk 10]
```diff
@@ -2,17 +2,54 @@
 
 =head1 NAME
 
-DTI --- A set of utility functions for performing common tasks relating to DTI data (particularly with regards to perform DTI QC)
+DTI::DTI --- A set of utility functions for performing common tasks relating to
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply DTIPrep/DTI/DTI.pm]: done too! Hope I did not miss anything.

**Comment by cmadjar** [comment]
@nicolasbrossard I think I integrated all your comments. Thanks!!

**Comment by cmadjar** [comment]
@nicolasbrossard Ready for another round of review :)

**Pull request merged**